Loggie/Classes/LogList/LogListTableViewController.swift

```diff
@@ -50,29 +59,73 @@ public class LogListTableViewController: UITableViewController {
         tableView.register(cellNib, forCellReuseIdentifier: LogListTableViewController.cellReuseIdentifier)
     }
 
-    // MARK: - Table view data source
+    @IBAction func clearAction() {
+        logsDataSourceDelegate.clearLogs()
+        logs = []
+        searchResult = []
+    }
+}
+
+// MARK: - Table view data source
+
+extension LogListTableViewController {
 
     override public func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
 
     override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return logs.count
+        return dataSource.count
     }
 
     override public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: LogListTableViewController.cellReuseIdentifier, for: indexPath) as! LogListTableViewCell
-        cell.configure(with: logs[indexPath.row])
+        let cell = tableView.dequeueReusableCell(
+            withIdentifier: LogListTableViewController.cellReuseIdentifier,
+            for: indexPath
+        ) as! LogListTableViewCell
+
+        cell.configure(with: dataSource[indexPath.row])
         return cell
     }
+}
+
+// MARK: - Table view delegate
 
-    // MARK: - Table view delegate
+extension LogListTableViewController {
 
     override public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        showLogDetails(with: logs[indexPath.row])
+        showLogDetails(with: dataSource[indexPath.row])
+    }
+
+    public override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        if searchBar.isFirstResponder {
+            searchBar.resignFirstResponder()
+        }
     }
+}
+// MARK: - Search bar delegate
+
+extension LogListTableViewController: UISearchBarDelegate {
+
+    public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        isSearching = searchText.isEmpty == false
```

Comment:
I would suggest extracting this to some method, maybe called `filterLogs` and call that method on `textDidChange` and on loggieDidUpdateLogs.

Reply:
As a matter of fact, there is already method `updateLogs`, just append this logic to that method, especially since we have publicly exposed filter parameter where client can additionally filter some logs. With that change you'll gain autoupdating list - if you have some search keyword entered, then the new logs will be filtered as well.